astropy/coordinates/representation.py

```diff
@@ -106,7 +106,7 @@ def unit(self):
         if self._parent is None:
             return None
 
-        unit = self._parent._unitstr
+        unit = self._parent._unitstr or "dimensionless"
```

Comment:
The one annoyance with this is that it cannot be parsed as a unit - but perhaps that is only a reason to enable that...

Reply:
Hmm. Is that a problem for table I/O ?
I presume that would be caught in a test, but I haven't checked.

Reply:
No, here it is purely for display. Overall, I would not worry about it - maybe eventually we'll decide upon a standard string and then allow that to be parsed.

Reply:
I'm just wondering about the need for `dimensionless`. Right now in Table the convention is that if you don't see an explicit unit, then it is dimensionless. For a normal `Quantity` with no unit in a `QTable`, the `unit` is shown as a blank. With this change there is an inconsistent behavior between `Quantity` and `RepresentationOrDifferential`.

Reply:
Now seeing #9588, I guess the other option is to change the displayed unit for `Quantity` to be `dimensionless` for consistency. This PR does not actually fix #9588 right now.

Reply:
(Unresolving) I somewhat agree with @taldcroft - maybe silly, but how about separating this out from the part that we all want to have, which is to mention the class in the header? We could then try solving "dimensionless" for all types that have units.